### DISCUSSION TITLE
Warning about non-page component, which shouldn't be displayed

### DISCUSSION THREAD

**Discussion by openscript** [gatsbyjs/gatsby#36052] [Help]
I'm generating pages from my gatsby-node.ts and I have multiple templates, which I pass to createPage({component: resolve(...), ...});. Some of the templates are not used yet, as the users didn't create any pages with it.

During the build Gatsby shows the following warning:

```
warn The GraphQL query in the non-page component "/workspaces/project/src/templates/GenericPageWithGallery.tsx" will not be run.
Exported queries are only executed for Page components. It's possible you're
trying to create pages in your gatsby-node.js and that's failing for some
reason.

If the failing component(s) is a regular component and not intended to be a page
component, you generally want to use a <StaticQuery> (https://gatsbyjs.com/docs/static-query)
instead of exporting a page query.

If you're more experienced with GraphQL, you can also export GraphQL
fragments from components and compose the fragments in the Page component
query and pass data down into the child component — https://graphql.org/learn/queries/#fragments
```

As soon as all templates are used the warning disappears. It's a bit confusing. Is this behavior intended or should I create an issue?

**Comment by jongwooo** [comment]
@openscript You get this warning when template isn't used anywhere in the `gatsby-node.ts`.

**Reply by openscript** [reply]
Thank you for answering. Yes, I found this out too, but couldn't it be, that a template is assigned dynamically and used later by a content manager changing the type of a page.

**Comment by KyleAMathews** [comment]
What's wrong with the warning? It seems like it's accurate then in your situation? It is a warning not an error as there's cases like yours where sometimes a template is intentionally not used.

**Reply by openscript** [reply]
Thank you for answering. Nothing is wrong with a warning per se, but the warning is not immediately clear or doesn't cover the case, if a template is just not used currently. 

My suggestion is to change the first paragraph to (last sentence):

```
warn The GraphQL query in the non-page component "/workspaces/project/src/templates/GenericPageWithGallery.tsx" will not be run.
Exported queries are only executed for Page components. It's possible you're trying to create pages in your gatsby-node.js and that's failing for some reason or it's not currently used.
```

Should I sent a PR for this?
